File: binance4j-websocket/src/main/java/com/binance4j/websocket/dto/DepthLevel.java
package com.binance4j.websocket.dto;

/** The order book size */
public enum DepthLevel {
	/** Depth of 5 results. */
	LEVEL_5("5"),
	/** Depth of 10 results. */
	LEVEL_10("10"),
	/** Depth of 20 results. */
	LEVEL_20("20");

	/** The result size. */
	final String value;

	/**
	 * @param value
	 */
	private DepthLevel(String value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return value;
	}
}

File: src/main/kotlin/com/binance4j/connectors/websocket/dto/DepthLevel.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.websocket.dto

import com.fasterxml.jackson.annotation.JsonProperty

/** The order book size.
 * @property value Result size. */
enum class DepthLevel(@JsonProperty("value")
	var value: String) {
	/** Depth of 5 results.  */
	LEVEL_5("5"),
	
	/** Depth of 10 results.  */
	LEVEL_10("10"),
	
	/** Depth of 20 results.  */
	LEVEL_20("20");
	
	override fun toString(): String = value
}